sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobCheckpointStoreInternal.cs

```diff
@@ -204,6 +204,30 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
                     // Even though documentation states otherwise, we cannot use UploadAsync when the blob already exists in
                     // the current storage SDK.  For this reason, we are using the specified ETag as an indication of what
                     // method to use.
+                    //
+                    // If an ETag is associated with ownership, assume the blob exists and attempt to update it.
+
+                    if (ownership.Version != null)
+                    {
+                        try
+                        {
+                            blobRequestConditions.IfMatch = new ETag(ownership.Version);
+                            infoResponse = await blobClient.SetMetadataAsync(metadata, blobRequestConditions, cancellationToken).ConfigureAwait(false);
+
+                            ownership.LastModifiedTime = infoResponse.Value.LastModified;
+                            ownership.Version = infoResponse.Value.ETag.ToString();
+                        }
+                        catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.BlobNotFound)
```

Comment:
Do we handle the case where multiple partition workers encounter this and race to upload the blob?